openedx/core/djangoapps/notifications/tasks.py

```diff
@@ -38,3 +41,32 @@ def create_course_notification_preferences_for_courses(self, course_ids):
             f'Newly created course preferences: {newly_created}.\n'
         )
     logger.info('Completed task create_course_notification_preferences')
+
+
+@shared_task(ignore_result=True)
+@set_code_owner_attribute
+def delete_expired_notifications():
+    """
+    This task deletes all expired notifications
+    """
+    batch_size = settings.EXPIRED_NOTIFICATIONS_DELETE_BATCH_SIZE
+    expiry_date = datetime.now(UTC) - timedelta(days=settings.NOTIFICATIONS_EXPIRY)
+    logger.info(f'Deleting expired notifications with batch size: {batch_size}')
+    start_time = datetime.now()
+    total_deleted = 0
+    delete_count = None
+    while delete_count != 0:
+        batch_start_time = datetime.now()
```

Comment:
Can it be done using LIMIT clause in delete? 
Ref: https://www.techonthenet.com/mysql/delete_limit.php

Reply:
DELETE with LIMIT is not supported in this version of MySQL... 

Using 
```
Notifications.object.filter(created__lte=expiry_date)[:batch_size].delete()
```
will use LIMIT with DELETE and it raises AssertionError.